### PR TITLE
Fix multiprocessing for modern Python

### PR DESCRIPTION
**Pull request by Eric-Arellano**
With Py39+, I'm getting this error running tests:

```
      File "/Users/arellano/code/qiskit-bot/qiskit_bot/release_process.py", line 284, in finish_release
    multiprocessing.Process(target=_meta_process).start()

      File "/Users/arellano/.pyenv/versions/3.9.16/lib/python3.9/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)

      File "/Users/arellano/.pyenv/versions/3.9.16/lib/python3.9/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)

      File "/Users/arellano/.pyenv/versions/3.9.16/lib/python3.9/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)

      File "/Users/arellano/.pyenv/versions/3.9.16/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)

      File "/Users/arellano/.pyenv/versions/3.9.16/lib/python3.9/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)

      File "/Users/arellano/.pyenv/versions/3.9.16/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)

      File "/Users/arellano/.pyenv/versions/3.9.16/lib/python3.9/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)

    AttributeError: Can't pickle local object 'finish_release.<locals>._meta_process'
```

(Didn't try Py38 because I can't on an M1).

This is because local functions are not picklable. It sounds like Python 3.8+, the default forking strategy changed from `fork` to `spawn`, which may explain why this has been passing in CI. 

The fix is to move the function to the top-level and then use `functools.partial` (partial application) to set all of its arguments before passing to `multiprocessing`.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9, "3.10"]
```
Comment (mtreinish) [inline .github/workflows/main.yml]: This is actually something we'll need to address on the deployed version. I just checked the deployed version of the bot and we're running it in Python 3.7. So we'll have to upgrade the docker image's python version on the deployment side to use a newer python version.

Comment (mtreinish) [inline .github/workflows/main.yml]: Is there something that doesn't work with 3.11 here?

**Review by mtreinish — COMMENTED** [review]
This LGTM, I forget that other OSes use spawn for multiprocessing instead of fork which is the source of this issue. The only question I have is whether we want to add python 3.11 here too, but otherwise the code looks good, thanks for fixing this.

File: .github/workflows/main.yml (hunk 1 above)
Comment (Eric-Arellano) [reply .github/workflows/main.yml]: Maybe we revert this CI change here? It was bad code hygiene on my part to mix a CI upgrade with the fix in qiskit_bot.

Comment (Eric-Arellano) [reply .github/workflows/main.yml]: I didn't try

**Comment by Eric-Arellano** [comment]
Bump @mtreinish

**Review by mtreinish — APPROVED** [review]

**Pull request merged**